pennylane/operation.py

```diff
@@ -300,48 +395,68 @@ def hash(self):
         """int: returns an integer hash uniquely representing the operator"""
         return hash((str(self.name), tuple(self.wires.tolist()), _process_data(self)))
 
-    @classmethod
-    def _matrix(cls, *params):
-        """Matrix representation of the operator
-        in the computational basis.
+    # pylint:disable=unused-argument
+    @staticmethod
+    def compute_matrix(*params, **hyperparams):
+        """Basic matrix representation of this operator in the computational basis.
 
-        This is a *class method* that should be defined for all
-        new operations and observables, that returns the matrix representing
-        the operator in the computational basis.
+        **Examples:**
 
-        This private method allows matrices to be computed
-        directly without instantiating the operators first.
+        >>> qml.CNOT.compute_matrix()
+        [[1, 0, 0, 0],
+        [0, 1, 0, 0],
+        [0, 0, 0, 1],
+        [0, 0, 1, 0]]
 
-        To return the matrices of *instantiated* operators,
-        please use the :attr:`~.Operator.matrix` property instead.
+        The matrix representation may depend on parameters or hyperparameters:
 
-        **Example:**
+        >>> qml.Rot.compute_matrix(0.1, 0.2, 0.3)
+        [[ 0.97517033-0.19767681j -0.09933467+0.00996671j]
+         [ 0.09933467+0.00996671j  0.97517033+0.19767681j]]
 
-        >>> qml.RY._matrix(0.5)
-        >>> array([[ 0.96891242+0.j, -0.24740396+0.j],
-                   [ 0.24740396+0.j,  0.96891242+0.j]])
+        Args:
```

Comment:
as per our current conventions this should be moved above `Example`

Reply:
Oh sorry, I forgot about that!